src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -432,6 +433,10 @@ public function getInstanceofConditionals()
      */
     public function setAutoconfigured($autoconfigured)
     {
+        if (null !== $this->decoratedService && $autoconfigured) {
```

Comment:
Maybe should we also throw in `setDecoratedService` is `$autoconfigured` is `true`?